File: WoloxAndroidBootstrap/app/src/main/java/ar/com/wolox/android/presenter/LeView.java
package ar.com.wolox.android.presenter;

/**
 * Interface to extend from the views that have a loading view along with an error one.
 */
public interface LeView {

    /**
     * Display a loading view while loading data in background.
     * <b>The loading view must have the id = R.id.loadingView</b>
     */
    void showLoading();


    /**
     * Show the error view.
     * <b>The error view must be a TextView with the id = R.id.errorView</b>
     *
     * @param e The Throwable that has caused this error
     */
    void showError(Throwable e);

}

File: core/src/main/java/ar/com/wolox/wolmo/core/presenter/LeView.kt
/*
 * Copyright (c) Wolox S.A
 *
 * Permission is hereby granted, free of charge, to any person obtaining a copy
 * of this software and associated documentation files (the "Software"), to deal
 * in the Software without restriction, including without limitation the rights
 * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
 * copies of the Software, and to permit persons to whom the Software is
 * furnished to do so, subject to the following conditions:
 *
 * The above copyright notice and this permission notice shall be included in
 * all copies or substantial portions of the Software.
 *
 * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
 * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
 * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
 * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
 * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
 * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
 * THE SOFTWARE.
 */
package ar.com.wolox.wolmo.core.presenter

/**
 * Interface to extend from the views that have a loading view along with an error one.
 */
interface LeView {

    /**
     * Display a loading view while loading data in background.
     * **The loading view must have the id = R.id.loadingView**
     */
    fun showLoading()


    /**
     * Show the error view.
     * **The error view must be a TextView with the id = R.id.errorView*
     * [e] is the Throwable that has caused this error
     */
    fun showError(e: Throwable)

}